src/dioptra/client/_client.py

```diff
@@ -0,0 +1,707 @@
+
+import requests
+
+
+class DioptraClient(object):
+    """Connects to the Dioptra REST api, and provides access to endpoints.
+
+    Args:
+        address: Address of the Dioptra REST api or if no address is given the DIOPTRA_RESTAPI_URI environment variable is used.
+
+    Notes:
+           See https://pages.nist.gov/dioptra/user-guide/api-reference-restapi.html for more
+           information on Dioptra's REST api.
+    """  # noqa B950
+
+    def __init__(self, address: str | None = None) -> None:
+        address = (
+            f"{address}/api" if address else f"{os.environ['DIOPTRA_RESTAPI_URI']}/api"
+        )
+        self._scheme, self._netloc, self._path, _, _, _ = urlparse(address)
+
+    @property
+    def experiment_endpoint(self) -> str:
+        """Experiment endpoint url"""
+        return urlunparse(
+            (self._scheme, self._netloc, urljoin(self._path, "experiment/"), "", "", "")
+        )
+
+    @property
+    def job_endpoint(self) -> str:
+        """Job endpoint url"""
+        return urlunparse(
+            (self._scheme, self._netloc, urljoin(self._path, "job/"), "", "", "")
+        )
+
+    @property
+    def task_plugin_endpoint(self) -> str:
+        """Task plugins endpoint url"""
+        return urlunparse(
+            (self._scheme, self._netloc, urljoin(self._path, "taskPlugin/"), "", "", "")
+        )
+
+    @property
+    def task_plugin_builtins_endpoint(self) -> str:
+        """Builtin task plugins endpoint url"""
+        return urlunparse(
+            (
+                self._scheme,
+                self._netloc,
+                urljoin(self._path, "taskPlugin/dioptra_builtins"),
+                "",
+                "",
+                "",
+            )
+        )
+
+    @property
+    def task_plugin_custom_endpoint(self) -> str:
+        """Custom task plugins endpoint url"""
+        return urlunparse(
+            (
+                self._scheme,
+                self._netloc,
+                urljoin(self._path, "taskPlugin/dioptra_custom"),
+                "",
+                "",
+                "",
+            )
+        )
+
+    @property
+    def queue_endpoint(self) -> str:
+        """Queue endpoint url"""
+        return urlunparse(
+            (self._scheme, self._netloc, urljoin(self._path, "queue/"), "", "", "")
+        )
+
+    def delete_custom_task_plugin(self, name: str) -> dict[str, Any]:
+        """Deletes a custom task plugin by its unique name.
+
+        Args:
+            name: A unique string identifying a task plugin package within dioptra_custom collection.
+
+        Returns:
+            The Dioptra REST api's response.
+
+            Example::
+
+                {'collection': 'dioptra_custom',
+                'status': 'Success',
+                'taskPluginName': ['evaluation']}
+
+        Notes:
+           See https://pages.nist.gov/dioptra/user-guide/api-reference-restapi.html for more
+           information on Dioptra's REST api.
```

Comment:
Ensure that all docstring indents are in 4-space increments.

```suggestion
        See https://pages.nist.gov/dioptra/user-guide/api-reference-restapi.html for
        more information on Dioptra's REST api.
```